contracts/asset-proxy/contracts/src/bridges/DydxBridge.sol

```diff
@@ -0,0 +1,239 @@
+                // Withdraw tokens from dydx to the `otherAccount`.
+                actions[i] = _createWithdrawAction(
+                    withdrawTo,
+                    scaledAmount,
+                    bridgeAction
+                );
+            } else {
+                // If all values in the `Action` enum are handled then this
+                // revert is unreachable: Solidity will revert when casting
+                // from `uint8` to `Action`.
+                revert("DydxBridge/UNRECOGNIZED_BRIDGE_ACTION");
+            }
+        }
+    }
+
+    /// @dev Returns a dydx `DepositAction`.
+    /// @param depositFrom Deposit tokens from this address who is also the account owner.
+    /// @param amount of tokens to deposit.
+    /// @param bridgeAction A `BridgeAction` struct.
+    /// @return depositAction The encoded dydx action.
+    function _createDepositAction(
+        address depositFrom,
+        uint256 amount,
+        BridgeAction memory bridgeAction
+    )
+        internal
+        pure
+        returns (
+            IDydx.ActionArgs memory depositAction
+        )
+    {
+        // Create dydx amount.
+        IDydx.AssetAmount memory dydxAmount = IDydx.AssetAmount({
+            sign: true,                                 // true if positive.
+            denomination: IDydx.AssetDenomination.Wei,  // Wei => actual token amount held in account.
+            ref: IDydx.AssetReference.Target,           // Target => an absolute amount.
+            value: amount                               // amount to deposit.
+        });
+
+        // Create dydx deposit action.
+        depositAction = IDydx.ActionArgs({
+            actionType: IDydx.ActionType.Deposit,           // deposit tokens.
+            amount: dydxAmount,                             // amount to deposit.
+            accountId: bridgeAction.accountId,              // index in the `accounts` when calling `operate`.
+            primaryMarketId: bridgeAction.marketId,         // indicates which token to deposit.
+            otherAddress: depositFrom,                      // deposit from the account owner.
+            // unused parameters
+            secondaryMarketId: 0,
+            otherAccountId: 0,
+            data: hex''
+        });
+    }
+
+    /// @dev Returns a dydx `WithdrawAction`.
+    /// @param withdrawTo Withdraw tokens to this address.
+    /// @param amount of tokens to withdraw.
+    /// @param bridgeAction A `BridgeAction` struct.
+    /// @return withdrawAction The encoded dydx action.
+    function _createWithdrawAction(
+        address withdrawTo,
+        uint256 amount,
+        BridgeAction memory bridgeAction
+    )
+        internal
+        pure
+        returns (
+            IDydx.ActionArgs memory withdrawAction
+        )
+    {
+        // Create dydx amount.
+        IDydx.AssetAmount memory amountToWithdraw = IDydx.AssetAmount({
+            sign: true,                                     // true if positive.
```

Comment:
Could be wrong, but I was under the impression that this needed to be `false` for withdraws. Ref https://github.com/dydxprotocol/solo/blob/master/contracts/protocol/impl/OperationImpl.sol#L377

Reply:
I looked back at my test transactions and it looks like the encoding is different when operating on a portion of the balance vs the entire balance. This does need to be `false` for partial; I've updated this plus set the `ref` to `Delta` for each.

Reply:
Mm maybe their client differentiates in order to avoid rounding errors in the contract.